src/bls/bls.service.ts

```diff
@@ -0,0 +1,105 @@
+import * as blst from '@chainsafe/blst';
+import { WINSTON_MODULE_NEST_PROVIDER } from 'nest-winston';
+import { Type, fromHexString } from '@chainsafe/ssz';
+import { ProviderService } from 'provider';
+import {
+  Inject,
+  Injectable,
+  OnModuleInit,
+  LoggerService,
+} from '@nestjs/common';
+import {
+  BLSPubkey,
+  Bytes32,
+  DOMAIN_DEPOSIT,
+  GENESIS_FORK_VERSION_BY_CHAIN_ID,
+  UintNum64,
+  ZERO_HASH,
+} from './bls.constants';
+import { DepositMessage, ForkData, SigningData } from './bls.containers';
+import { DepositData } from './interfaces';
+
+@Injectable()
+export class BlsService implements OnModuleInit {
+  constructor(
+    @Inject(WINSTON_MODULE_NEST_PROVIDER) private logger: LoggerService,
+    private providerService: ProviderService,
+  ) {}
+
+  async onModuleInit() {
+    const chainId = await this.providerService.getChainId();
```

Comment:
This is place is a little weak because of the network is able to throw an exception.

Reply:
Ok. It's not clear that it works, you described.